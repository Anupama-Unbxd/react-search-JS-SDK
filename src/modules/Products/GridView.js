import React from 'react';
import PropTypes from 'prop-types';

import { GridProductCard } from './ProductCards';
//We need the fieldMap object to map to values

const GridView = (props) => {

    const { products = [],
        per_row,
        fieldMap,
        variantMap,
        unbxdProductCardClickHandler } = props;

    return (<div className={`Unbx-product-container Unbx-grid-view grid grid-cols-${per_row} gap-4 p-4`}>{
        products.map((product) => {
            return (<GridProductCard product={product}
                fieldMap={fieldMap}
                variantMap={variantMap}
                per_row={per_row}
                key={product.uniqueId}
                unbxdProductCardClickHandler={unbxdProductCardClickHandler} />)
        })

    }
    </div>)
}

GridView.propTypes = {
    products: PropTypes.arrayOf(PropTypes.object).isRequired,
    per_row: PropTypes.number.isRequired,
    fieldMap: PropTypes.object.isRequired,
    variantMap: PropTypes.object.isRequired,
    unbxdProductCardClickHandler: PropTypes.func.isRequired
}

export default GridView;
