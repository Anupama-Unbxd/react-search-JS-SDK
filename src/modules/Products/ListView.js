import React from 'react';
import PropTypes from 'prop-types';

import { ListProductCard } from './ProductCards';

const ListView = ({ products = [], fieldMap, isVariant, variantMap, unbxdProductCardClickHandler }) => {

    return (<div className={`Unbx-product-container Unbx-grid-view grid grid-cols-1 gap-4 p-4`}>{
        products.map((product) => {
            return (<ListProductCard product={product}
                fieldMap={fieldMap}
                isVariant={isVariant}
                variantMap={variantMap}
                key={product.uniqueId}
                unbxdProductCardClickHandler={unbxdProductCardClickHandler} />)
        })

    }
    </div>)
}
ListView.propTypes = {
    products: PropTypes.arrayOf(PropTypes.object).isRequired,
    per_row: PropTypes.number.isRequired,
    fieldMap: PropTypes.object.isRequired,
    isVariant: PropTypes.bool.isRequired,
    variantMap: PropTypes.object.isRequired,
    unbxdProductCardClickHandler: PropTypes.func.isRequired
}

export default ListView;
