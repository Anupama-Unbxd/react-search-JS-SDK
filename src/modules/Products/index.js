import React from 'react';
import AppContext, { AppContextConsumer } from '../../common/context'
import { ProductContextProvider } from './context'

import PropTypes from 'prop-types';

import ViewTypes from './ViewTypes';
import ProductsView from './ProductsView';
import { conditionalRenderer } from '../../common/utils';

class Products extends React.PureComponent {

    ProductContext = React.createContext();

    static contextType = AppContext;

    static ViewTypes = ViewTypes;
    static ProductsView = ProductsView;

    onViewToggle = (event) => {
        this.setState({ isGrid: event.target.id === 'Unbx-grid' })
    }

    constructor(props) {
        super(props);

        //Set all data that is required in state so that it can be passed down to children
        const { ZeroResultsTemplate, per_row, per_page,
            infiniteScroll, loadOnClick, fieldMap, variantMap } = this.props

        this.state = {
            isGrid: true,
            onViewToggle: this.onViewToggle,
            ZeroResultsTemplate,
            per_row,
            per_page,
            infiniteScroll,
            loadOnClick,
            fieldMap,
            variantMap
        }
    }

    componentDidMount() {

        const { helpers: { setProductConfiguration } } = this.context;

        const { per_page, fieldMap, variants,
            variantCount, variantMap } = this.props

        //generate required fields here based on fieldMap and variantFields
        const requiredFields = Object.values(fieldMap);
        const variantRequiredFields = Object.values(variantMap);

        //Set the main config with these local configs
        setProductConfiguration({
            per_page, requiredFields, variants, variantCount, variantRequiredFields
        });
    }

    render() {

        const { per_page } = this.state;

        return (<AppContextConsumer>
            {({ unbxdCore }) => {

                const getSearchResults = unbxdCore.getSearchResults.bind(unbxdCore);
                //Wrap this in a function so that there is only one function call
                const setPageStart = unbxdCore.setPageStart.bind(unbxdCore);
                const getPaginationInfo = unbxdCore.getPaginationInfo.bind(unbxdCore);
                const getResults = unbxdCore.getResults.bind(unbxdCore);

                //Get next page method
                const getNextPage = () => {

                    //Get the next batch by jumping the current per_page
                    const { currentPage } = getPaginationInfo();
                    const newStart = currentPage === 0 ? per_page : currentPage + per_page;
                    setPageStart(newStart);
                    getResults();
                }

                const unbxdProductCardClickHandler = () => {
                    console.log("hello from unbxdProductCardClickHandler")
                }

                //onClick for products
                //We also need to pass infinite scroll method
                const getProductsProp = ({ onClick, ...props }) => ({
                    onClick: () => {
                        unbxdProductCardClickHandler();
                        onClick();
                    },
                    getSearchResults,
                    getNextPage,
                    ...props
                })

                const DefaultRender = <React.Fragment>
                    <ViewTypes />
                    <ProductsView />
                </React.Fragment>

                //We will be using components here which can also be used as compound components
                //We need to send prop getters for render props. mainly onclick
                return (<ProductContextProvider value={{ ...this.state, getSearchResults, getNextPage, unbxdProductCardClickHandler }}>
                    {conditionalRenderer(this.props.children, {
                        state: this.state,
                        getProductsProp,
                    }, DefaultRender)}
                </ProductContextProvider>)
            }
            }
        </AppContextConsumer >)
    }
}

Products.defaultProps = {
    per_row: 5,
    per_page: 10,
    //requiredFields: ["imageUrl", "title", "price"],
    fieldMap: { productName: "title", uniqueId: "uniqueId", imageUrl: "imageUrl" },
    ZeroResultsTemplate: false,
    infiniteScroll: false,
    loadOnClick: false,
    variants: false,
}

Products.propTypes = {
    per_row: PropTypes.number,
    per_page: PropTypes.number,
    fieldMap: PropTypes.object.isRequired,
    ZeroResultsTemplate: PropTypes.oneOfType([PropTypes.element, PropTypes.bool, PropTypes.func]),
    infiniteScroll: PropTypes.bool,
    loadOnClick: PropTypes.bool,
    variants: PropTypes.bool,
    variantCount: PropTypes.number,
    variantMap: PropTypes.object,
    variantRequiredFields: PropTypes.arrayOf(PropTypes.string),
}

export default Products;
