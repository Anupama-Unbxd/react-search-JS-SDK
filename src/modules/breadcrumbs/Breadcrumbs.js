import React from 'react';
import PropTypes from 'prop-types';

import { AppContextConsumer } from '../../common/context';
import { hasUnbxdSearchWrapperContext } from '../../common/utils';
import BreadcrumbsContainer from './BreadcrumbsContainer';

/**
 * Component to manage the breadcrumb.
 */
const Breadcrumbs = (props) => {
    return (
        <AppContextConsumer>
            {(appState) => {
                if (appState === undefined) {
                    hasUnbxdSearchWrapperContext(Breadcrumbs.displayName);
                }

                const {
                    unbxdCore,
                    unbxdCoreStatus,
                    helpers,
                    productType,
                } = appState;

                return (
                    <BreadcrumbsContainer
                        unbxdCore={unbxdCore}
                        unbxdCoreStatus={unbxdCoreStatus}
                        productType={productType}
                        helpers={helpers}
                        {...props}
                    />
                );
            }}
        </AppContextConsumer>
    );
};

Breadcrumbs.displayName = 'Breadcrumbs';

Breadcrumbs.defaultProps = {};

Breadcrumbs.propTypes = {
    /**
     * Root component of the breadcrumb.
     */
    Root: PropTypes.oneOfType([
        PropTypes.element,
        PropTypes.func,
        PropTypes.node,
    ]),
    /**
     * Separator node of the breadcrumb.
     */
    separator: PropTypes.node,
    /**
     * Custom breadcrumb component.
     */
    BreadcrumbItemComponent: PropTypes.oneOfType([
        PropTypes.element,
        PropTypes.func,
    ]),
};

export default Breadcrumbs;
