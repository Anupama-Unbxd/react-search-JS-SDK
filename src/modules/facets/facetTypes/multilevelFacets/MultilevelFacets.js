import React, { Fragment } from 'react';

import { FacetsContextConsumer } from '../../context';
import GenerateFacets from './generateFacets/GenerateFacets';
import BreadCrumbs from './breadCrumbs/BreadCrumbs';

const MultilevelFacets = () => {

    return (<FacetsContextConsumer>
        {({ data, helpers }) => {

            const { getBucketedFacets,
                addCategoryFilter,
                removeCategoryFilter,
                getSelectedBucketedFacet,
                getBreadCrumbsList,
                MultilevelFacetItemComponent,
                BreadcrumbItemComponent } = helpers;

            const bucketedFacets = getBucketedFacets();
            const selectedBucketedFacet = getSelectedBucketedFacet();
            const breadCrumbsList = getBreadCrumbsList();

            return (<Fragment>

                <BreadCrumbs
                    breadCrumbsList={breadCrumbsList}
                    removeCategoryFilter={removeCategoryFilter}
                    BreadcrumbItemComponent={BreadcrumbItemComponent}
                />

                <GenerateFacets bucketedFacets={bucketedFacets}
                    selectedBucketedFacet={selectedBucketedFacet}
                    addCategoryFilter={addCategoryFilter}
                    breadCrumbsList={breadCrumbsList}
                    MultilevelFacetItemComponent={MultilevelFacetItemComponent}
                />
            </Fragment>)
        }
        }</FacetsContextConsumer>)

}

export default MultilevelFacets;
