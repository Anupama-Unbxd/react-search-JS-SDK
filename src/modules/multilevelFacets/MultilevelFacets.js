import React from 'react';
import PropTypes from 'prop-types';

import { AppContextConsumer } from '../../common/context';
import { hasUnbxdSearchWrapperContext } from '../../common/utils';
import MultilevelFacetsContainer from './MultilevelFacetsContainer';

/**
 * Component to manage multilevel categories.
 */
const MultilevelFacets = (props) => {
    return (
        <AppContextConsumer>
            {(appState) => {
                if (appState === undefined) {
                    hasUnbxdSearchWrapperContext(MultilevelFacets.displayName);
                }

                const { unbxdCore, unbxdCoreStatus, helpers } = appState;

                return (
                    <MultilevelFacetsContainer
                        unbxdCore={unbxdCore}
                        unbxdCoreStatus={unbxdCoreStatus}
                        helpers={helpers}
                        {...props}
                    />
                );
            }}
        </AppContextConsumer>
    );
};

MultilevelFacets.displayName = 'MultilevelFacets';

MultilevelFacets.defaultProps = {
    defaultCategoryFilter: '',
    facetDepth: 6,
    facetLimit: 100,
    collapsible: false,
    searchable: false,
};

MultilevelFacets.propTypes = {
    /**
     * Display name of the category.
     */
    categoryDisplayName: PropTypes.string.isRequired,
    /**
     * Category field in the feed.
     */
    categoryField: PropTypes.string.isRequired,
    /**
     * Maximum depth of the category.
     */
    facetDepth: PropTypes.number,
    /**
     * Maximum number of values present in a facet.
     */
    facetLimit: PropTypes.number,
    /**
     * Default category filter
     */
    defaultCategoryFilter: PropTypes.string,
    /**
     * Collapse facet values
     */
    collapsible: PropTypes.bool,
    /**
     * Search facet values
     */
    searchable: PropTypes.bool,
    /**
     * Custom Multilevel facet component
     */
    FacetItemComponent: PropTypes.oneOfType([
        PropTypes.element,
        PropTypes.func,
    ]),
    /**
     * Label for the component.
     */
    label: PropTypes.node,
    /**
     * Callback for facet click.
     */
    onFacetClick: PropTypes.node,
};

export default MultilevelFacets;
