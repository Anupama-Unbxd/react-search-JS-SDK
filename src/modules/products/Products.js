import React from 'react';
import PropTypes from 'prop-types';

import AppContext from '../../common/context'
import { ProductContextProvider } from './context'
import ViewTypes from './viewTypes/ViewTypes';
import ProductsView from './productsView/ProductsView';
import { conditionalRenderer, hasUnbxdSearchWrapperContext } from '../../common/utils';
import { Loader as defaultLoader } from '../../components'
import { getProductViewType } from './utils';
import { trackProductClick } from '../analytics';


/**
 * Component to render the search products. 
 * Products supports Product View types `LIST` and `GRID` out of the box. 
 * Products also manages the pagination options for the search results here.
 */
export class Products extends React.PureComponent {

    constructor(props) {
        super(props);

        const { productViewTypes } = this.props

        //Lets fallback to grid view
        const productViewType = getProductViewType(productViewTypes)[0];

        this.state = {
            productViewType,
        }
    }

    componentDidMount() {

        if (this.context === undefined) {
            hasUnbxdSearchWrapperContext(Products.displayName);
        }

        const { helpers: { setProductConfiguration } } = this.context;

        const { pageSize,
            attributesMap,
            showVariants,
            variantsCount,
            variantAttributesMap,
            paginationType,
            groupBy } = this.props;

        //generate required fields here based on attributesMap and variantAttributesMap
        const requiredFields = Object.values(attributesMap);
        const variantRequiredFields = Object.values(variantAttributesMap);

        //Set the main config
        //Do not set pageSize if pagination type is FIXED_SCROLLING
        setProductConfiguration({
            pageSize,
            requiredFields,
            showVariants,
            variantsCount,
            variantRequiredFields,
            paginationType,
            groupBy
        });
    }

    getProductProps() {

        if (this.context === undefined) {
            hasUnbxdSearchWrapperContext(Products.displayName);
        }

        const { unbxdCore, unbxdCoreStatus, helpers: { trackActions } } = this.context;
        const { ZeroResultsComponent,
            perRow,
            pageSize,
            paginationType,
            productViewTypes,
            productViewDisplayType,
            heightDiffToTriggerNextPage,
            LoaderComponent,
            showLoader,
            onProductClick: onProductClickCB,
            LoadMoreComponent,
            onZeroResults,
            ProductItemComponent,
            ProductsViewItemComponent,
            productIdAttribute,
            attributesMap,
            showVariants,
            groupBy,
            variantAttributesMap,
            showSwatches,
            swatchAttributes,
            SwatchItemComponent } = this.props;

        const getSearchResults = unbxdCore.getSearchResults.bind(unbxdCore);
        const setPageStart = unbxdCore.setPageStart.bind(unbxdCore);
        const getPaginationInfo = unbxdCore.getPaginationInfo.bind(unbxdCore);
        const getResults = unbxdCore.getResults.bind(unbxdCore);
        const getProductByPropValue = unbxdCore.getProductByPropValue.bind(unbxdCore);

        const query = unbxdCore.getSearchQuery() || "";

        const onViewToggle = (event) => {

            const productViewType = event.target.dataset.viewtype || event.target.value;;

            this.setState({ productViewType });
        }

        const getNextPage = () => {

            const { currentPage } = getPaginationInfo();
            const newStart = currentPage === 0 ? pageSize : (currentPage * pageSize);

            setPageStart(newStart);
            getResults();
        }

        const onProductClick = (event) => {

            const productUniqueId = event.target.dataset.uniqueid;
            if (productUniqueId) {
                const prank = event.target.dataset.prank;
                const clickedProduct = getProductByPropValue(productIdAttribute, productUniqueId);

                if (onProductClickCB) {

                    if (onProductClickCB(clickedProduct)) {

                        trackProductClick(clickedProduct[productIdAttribute], prank);
                    } else {

                        event.preventDefault();
                    }
                } else {

                    trackProductClick(clickedProduct[productIdAttribute], prank);
                }

            }

        }

        const getOnProductsClickProps = ({ onClick, ...props }) => ({
            onClick: () => {
                onProductClick();
                onClick();
            },
            ...props
        })

        const data = {
            unbxdCoreStatus,
            perRow,
            paginationType,
            attributesMap,
            variantAttributesMap,
            productViewTypes,
            heightDiffToTriggerNextPage,
            showVariants,
            showLoader,
            productViewDisplayType,
            showSwatches,
            swatchAttributes,
            groupBy,
            query,
            productIdAttribute,
            ...this.state
        };
        const helpers = {
            ZeroResultsComponent,
            onViewToggle,
            getOnProductsClickProps,
            getNextPage,
            onProductClick,
            getSearchResults,
            getNextPage,
            LoaderComponent,
            onZeroResults,
            ProductItemComponent,
            ProductsViewItemComponent,
            SwatchItemComponent,
            LoadMoreComponent
        }

        return { data, helpers }

    }

    render() {

        const DefaultRender = <React.Fragment>
            <ViewTypes />
            <ProductsView />
        </React.Fragment>;
        const LoaderRender = null;


        return (<ProductContextProvider value={this.getProductProps()}>
            {conditionalRenderer(this.props.children, this.getProductProps(), DefaultRender, LoaderRender)}
        </ProductContextProvider>)
    }
}

Products.contextType = AppContext;
Products.ViewTypes = ViewTypes;
Products.ProductsView = ProductsView;
Products.displayName = "Products";

Products.defaultProps = {
    perRow: 4,
    pageSize: 10,
    productViewTypes: ["GRID"],
    attributesMap: {},
    variantAttributesMap: {},
    paginationType: 'FIXED_PAGINATION',
    heightDiffToTriggerNextPage: 100,
    showVariants: false,
    LoaderComponent: defaultLoader,
    showLoader: false,
    productViewDisplayType: "DROPDOWN",
    productIdAttribute: 'uniqueId',
}

Products.propTypes = {
    /**
    * Number of products to be shown per row.
    */
    perRow: PropTypes.number,
    /**
    * Number of products to be loaded on a page.
    */
    pageSize: PropTypes.number,
    /**
    * Required pagination type. Possible options are `INFINITE_SCROLL`, `CLICK_N_SCROLL` and `FIXED_PAGINATION`.
    */
    paginationType: PropTypes.string,
    /**
     * Height difference to trigger for next page in case of paginationType `INFINITE_SCROLL`.
     */
    heightDiffToTriggerNextPage: PropTypes.number,
    /**
     * Custom load more component for CLICK_N_SCROLL
     */
    LoadMoreComponent: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
    /**
    * Required ProductViewType.Possible options are`GRID` and `LIST`.
    */
    productViewTypes: PropTypes.arrayOf(PropTypes.string),
    /**
    * Required ProductViewType Display type.Possible options are`LIST` and `DROPDOWN`.
    */
    productViewDisplayType: PropTypes.string,
    /**
    * Custom `LIST` item component for product views.
    */
    ProductsViewItemComponent: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
    /**
    * Unique attribute of the product
    */
    productIdAttribute: PropTypes.string,
    /**
    * Mapping of catalog Product fields to SDK Product fields.
    */
    attributesMap: PropTypes.object.isRequired,
    /**
    * Show variants to a product.
    */
    showVariants: PropTypes.bool,
    /**
    * Number of variants to fetch.
    */
    variantsCount: PropTypes.number,
    /**
    * Mapping of catalog Product variant fields to SDK Product variant fields.
    */
    variantAttributesMap: PropTypes.object,
    /**
     * Should loader be shown
     */
    showLoader: PropTypes.bool,
    /**
    * Custom loader component
    */
    LoaderComponent: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
    /**
    * Callback function triggered on click of a product.
    */
    onProductClick: PropTypes.func,
    /**
    * Callback function triggered on zero results.
    */
    onZeroResults: PropTypes.func,
    /**
    * Display swatches
    */
    showSwatches: PropTypes.bool,
    /**
    * Group variants by the attribute
    */
    groupBy: PropTypes.string,
    /**
    * Swatch attributes that change on click on of the swatch
    */
    swatchAttributes: PropTypes.object,
    /**
    * Custom swatch component
    */
    SwatchItemComponent: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
    /**
    * Custom product item component
    */
    ProductItemComponent: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
    /**      
    *  Component to be shown in case of zero results.
    */
    ZeroResultsComponent: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
}

export default Products;
