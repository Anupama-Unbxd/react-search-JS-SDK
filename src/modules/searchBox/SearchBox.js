import React from 'react';
import PropTypes from 'prop-types';

import AppContext from '../../common/context'
import { SearchBoxContextProvider } from './context'
import SearchInput from './searchInput';
import SearchButton from './searchButton';
import { conditionalRenderer, isContext } from '../../common/utils';

/**
 * Component to set or update the search query.
 */
class SearchBox extends React.Component {

    constructor(props) {
        super(props);

        this.onQueryChange = this.onQueryChange.bind(this);
        this.onClearQuery = this.onClearQuery.bind(this);
        this.handleQuerySubmit = this.handleQuerySubmit.bind(this);

        this.state = { query: '' };
    }


    onQueryChange(event) {
        const query = event.target.value;
        this.setState({ query });
    }

    onClearQuery() {

        const { query } = this.state;
        const { onClear } = this.props;

        if (onClear) {
            onClear(query) && this.setState({ query: '' });
        } else {
            this.setState({ query: '' });
        }
    }

    handleQuerySubmit(event) {
        event.preventDefault();

        const { query } = this.state;
        const { onSubmit } = this.props;
        const { helpers: { setSearchBoxConfiguration, trackActions } } = this.context;

        if (onSubmit) {

            if (onSubmit(query) && query.length) {

                setSearchBoxConfiguration({ query });
                //track for search hit here
                trackActions({ type: 'SEARCH', data: { query } })
            }
        }
        else {

            if (query.length) {

                setSearchBoxConfiguration({ query });
                //track for search hit here
                trackActions({ type: 'SEARCH', data: { query } })
            }
        }


    }

    componentDidMount() {

        if (this.context === undefined) {
            isContext(Products.displayName);
        }
    }

    getSearchBoxProps() {

        const {
            isAutoFocus,
            isClear, } = this.props;

        const data = {
            isAutoFocus,
            isClear,
            ...this.state
        };
        const helpers = {
            onQueryChange: this.onQueryChange,
            handleQuerySubmit: this.handleQuerySubmit,
            onClearQuery: this.onClearQuery,

        };

        return { data, helpers };
    }


    render() {

        const DefaultRender = <form onSubmit={this.handleQuerySubmit} className='UNX-searchbox-container'>
            <SearchInput />
            <SearchButton />
        </form>

        return (<SearchBoxContextProvider value={this.getSearchBoxProps()}>
            {conditionalRenderer(this.props.children, this.getSearchBoxProps(), DefaultRender)}
        </SearchBoxContextProvider>)
    }
}

SearchBox.contextType = AppContext;

SearchBox.SearchInput = SearchInput;
SearchBox.SearchButton = SearchButton;

SearchBox.displayName = "SearchBox";

SearchBox.defaultProps = {
    isAutoFocus: false,
    isClear: false,
    onSubmit: null,
    onClear: null,
}

SearchBox.propTypes = {
    /**
    * Should the searchbox be focused by default.
    */
    isAutoFocus: PropTypes.bool,
    /**
    * Should the searchbox be clearable.
    */
    isClear: PropTypes.bool,
    /**
    * Hook for search query. The function should return `true` if the search is to be triggered, false otherwise.
    */
    onSubmit: PropTypes.func,
    /**
    * Hook for clearing the search query. The function should return `true` if the searchbox is to be cleared, false otherwise.
    */
    onClear: PropTypes.func,
}

export default SearchBox;
