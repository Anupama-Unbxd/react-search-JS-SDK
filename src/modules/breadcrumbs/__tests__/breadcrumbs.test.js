import renderer from 'react-test-renderer';
import { render, waitFor, fireEvent, screen } from '@testing-library/react';
import React from 'react';
import Products from '../../products/';
import Breadcrumbs from '../index';
import MultilevelFacets from '../../multilevelFacets';
import UnbxdSearchWrapper from '../../../UnbxdSearchWrapper';
import SearchBox from '../../searchBox';
import { breadcrumbSearchResponse } from './mocks/';

// establish API mocking before all tests
beforeAll(() => {
    window.fetch = jest.fn(() => {
        return Promise.resolve({
            json: () => Promise.resolve(breadcrumbSearchResponse),
        });
    });
})

const attributesMap = {
    productName: 'title',
    uniqueId: 'uniqueId',
    imageUrl: 'imageUrl',
    price: 'min_cheapest_default_price',
    productUrl: 'productUrl'
};

const Root = () => <span className="UNX-breadcrumb__root">Home</span>;
const separator = <span className="UNX-breadcrumb__separator">/</span>;

test('Match Snapshot for breadcrumbs', async () => {
    const tree = renderer
        .create(
            <UnbxdSearchWrapper
                siteKey="wildearthclone-neto-com-au808941566310465"
                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
            >
                <Breadcrumbs root={<Root />} separator={separator} />
                <Products attributesMap={attributesMap} />
                <div>
                    <SearchBox defaultSearch="shoes" />
                </div>
            </UnbxdSearchWrapper>
        );
    await waitFor(() => tree.toJSON());
    expect(tree.toJSON()).toMatchSnapshot();
});

test('Breadcrumbs test', async () => {
    const { getByText } = render(
        <>
            <UnbxdSearchWrapper
                siteKey="wildearthclone-neto-com-au808941566310465"
                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
            >
                <Breadcrumbs root={<Root />} separator={separator} />
                <MultilevelFacets />
                <Products attributesMap={attributesMap} />
                <div>
                    <SearchBox defaultSearch="shoes" />
                </div>
            </UnbxdSearchWrapper>
        </>
    );

    await waitFor(() => expect(getByText("SUBCATEGORY")).toBeInTheDocument());
    fireEvent.click(getByText("Sets"));
    await waitFor(() => {
        screen.findByText("Home");
    })
});
