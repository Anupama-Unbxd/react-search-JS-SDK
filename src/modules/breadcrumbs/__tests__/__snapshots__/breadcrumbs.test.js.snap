// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Snapshot for breadcrumbs 1`] = `
Array [
  <div
    className="UNX-breadcrumbs__container"
  >
    <div
      className="UNX-breadcrumbs__list"
    />
  </div>,
  <div
    className="UNX-breadcrumbs__container"
  >
    <div
      className="UNX-breadcrumbs__list"
    />
  </div>,
  <div>
    <div
      className="UNX-products__container"
    >
      <div
        className="UNX-products__list -grid grid-cols-4"
      >
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            data-prank={1}
            data-uniqueid="T624835"
            onClick={[Function]}
          >
            <img
              className="-image"
              data-prank={1}
              data-uniqueid="T624835"
            />
          </a>
          <div
            className="UNX-swatch__container"
          >
            <div
              className="UNX-swatch__list"
            />
          </div>
          <div
            className="-details"
          >
            <div
              className="-title"
              data-prank={1}
              data-uniqueid="T624835"
            />
            <div
              className="-price"
              data-prank={1}
              data-uniqueid="T624835"
            />
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            data-prank={2}
            data-uniqueid="T624611"
            onClick={[Function]}
          >
            <img
              className="-image"
              data-prank={2}
              data-uniqueid="T624611"
            />
          </a>
          <div
            className="UNX-swatch__container"
          >
            <div
              className="UNX-swatch__list"
            />
          </div>
          <div
            className="-details"
          >
            <div
              className="-title"
              data-prank={2}
              data-uniqueid="T624611"
            />
            <div
              className="-price"
              data-prank={2}
              data-uniqueid="T624611"
            />
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            data-prank={3}
            data-uniqueid="T624610"
            onClick={[Function]}
          >
            <img
              className="-image"
              data-prank={3}
              data-uniqueid="T624610"
            />
          </a>
          <div
            className="UNX-swatch__container"
          >
            <div
              className="UNX-swatch__list"
            />
          </div>
          <div
            className="-details"
          >
            <div
              className="-title"
              data-prank={3}
              data-uniqueid="T624610"
            />
            <div
              className="-price"
              data-prank={3}
              data-uniqueid="T624610"
            />
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            data-prank={4}
            data-uniqueid="T624604"
            onClick={[Function]}
          >
            <img
              className="-image"
              data-prank={4}
              data-uniqueid="T624604"
            />
          </a>
          <div
            className="UNX-swatch__container"
          >
            <div
              className="UNX-swatch__list"
            />
          </div>
          <div
            className="-details"
          >
            <div
              className="-title"
              data-prank={4}
              data-uniqueid="T624604"
            />
            <div
              className="-price"
              data-prank={4}
              data-uniqueid="T624604"
            />
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            data-prank={5}
            data-uniqueid="T623252"
            onClick={[Function]}
          >
            <img
              className="-image"
              data-prank={5}
              data-uniqueid="T623252"
            />
          </a>
          <div
            className="UNX-swatch__container"
          >
            <div
              className="UNX-swatch__list"
            />
          </div>
          <div
            className="-details"
          >
            <div
              className="-title"
              data-prank={5}
              data-uniqueid="T623252"
            />
            <div
              className="-price"
              data-prank={5}
              data-uniqueid="T623252"
            />
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            data-prank={6}
            data-uniqueid="T613177"
            onClick={[Function]}
          >
            <img
              className="-image"
              data-prank={6}
              data-uniqueid="T613177"
            />
          </a>
          <div
            className="UNX-swatch__container"
          >
            <div
              className="UNX-swatch__list"
            />
          </div>
          <div
            className="-details"
          >
            <div
              className="-title"
              data-prank={6}
              data-uniqueid="T613177"
            />
            <div
              className="-price"
              data-prank={6}
              data-uniqueid="T613177"
            />
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            data-prank={7}
            data-uniqueid="T613176"
            onClick={[Function]}
          >
            <img
              className="-image"
              data-prank={7}
              data-uniqueid="T613176"
            />
          </a>
          <div
            className="UNX-swatch__container"
          >
            <div
              className="UNX-swatch__list"
            />
          </div>
          <div
            className="-details"
          >
            <div
              className="-title"
              data-prank={7}
              data-uniqueid="T613176"
            />
            <div
              className="-price"
              data-prank={7}
              data-uniqueid="T613176"
            />
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            data-prank={8}
            data-uniqueid="ABI"
            onClick={[Function]}
          >
            <img
              className="-image"
              data-prank={8}
              data-uniqueid="ABI"
            />
          </a>
          <div
            className="UNX-swatch__container"
          >
            <div
              className="UNX-swatch__list"
            />
          </div>
          <div
            className="-details"
          >
            <div
              className="-title"
              data-prank={8}
              data-uniqueid="ABI"
            />
            <div
              className="-price"
              data-prank={8}
              data-uniqueid="ABI"
            />
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            data-prank={9}
            data-uniqueid="AACA"
            onClick={[Function]}
          >
            <img
              className="-image"
              data-prank={9}
              data-uniqueid="AACA"
            />
          </a>
          <div
            className="UNX-swatch__container"
          >
            <div
              className="UNX-swatch__list"
            />
          </div>
          <div
            className="-details"
          >
            <div
              className="-title"
              data-prank={9}
              data-uniqueid="AACA"
            />
            <div
              className="-price"
              data-prank={9}
              data-uniqueid="AACA"
            />
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            data-prank={10}
            data-uniqueid="710010105"
            onClick={[Function]}
          >
            <img
              className="-image"
              data-prank={10}
              data-uniqueid="710010105"
            />
          </a>
          <div
            className="UNX-swatch__container"
          >
            <div
              className="UNX-swatch__list"
            />
          </div>
          <div
            className="-details"
          >
            <div
              className="-title"
              data-prank={10}
              data-uniqueid="710010105"
            />
            <div
              className="-price"
              data-prank={10}
              data-uniqueid="710010105"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="UNX-searchbox__container"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          className="UNX-searchbox__input"
          onChange={[Function]}
          placeholder=""
          value="*"
        />
        <button
          className="UNX-searchbox__button"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
  </div>,
]
`;
